daml-foundations/daml-ghc/daml-compiler/src/DA/Service/Daml/Compiler/Impl/Handle.hs

```diff
@@ -210,60 +187,40 @@ buildDar service file mbExposedModules pkgName sdkVersion buildDataFiles dalfInp
         []
         pkgName
         sdkVersion
-    else do
-      pkg <- compileFile service file
+    else actionToExceptT service $ runMaybeT $ do
+      pkg <- useE GeneratePackage file
       let pkgModuleNames = S.fromList $ map T.unpack $ LF.packageModuleNames pkg
       let missingExposed = S.fromList (fromMaybe [] mbExposedModules) S.\\ pkgModuleNames
       unless (S.null missingExposed) $ do
-          liftIO $ CompilerService.logSeriousError service $
+          liftIO $ logSeriousError service $
               "The following modules are declared in exposed-modules but are not part of the DALF: " <>
               T.pack (show $ S.toList missingExposed)
-          throwE []
+          fail ""
```

Comment:
Why do we fail without any useful message here?

Reply:
We already emit the log message. There isn’t really any point in outputting it via diagnostics as well. And we are in MaybeT so the string here doesn’t matter either way.

Reply:
I considered `MaybeT (pure Nothing)` but that looks awkward. `empty` is nice but also somewhat unclear so `fail ""` seemed like the best option.

Reply:
I was about to add a github issue for adding something to `transformers`, or even a PR, but then I saw that it's on hub.darcs.net, which is a sign the project is so unmaintained i might as well not bother.